mantidproject/mantid
# Refactor context in muon GUI

**Description of work.**

This PR refactors the new FDA code to replace the context with a parent context and several sub contexts. This is to enable additions planned going forward easier and to maintain the code base in a good shape.

**To test:**

Open the new frequency domain analysis GUI and try loading and processing some date. It is also worth having quite a thorough check that all the options still work.



partially fixes #25306 

*This does not require release notes* because it is all internal refactoring


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
